### DISCUSSION TITLE
Parcel and elm-tooling

### DISCUSSION THREAD

**Discussion by emthornber** [parcel-bundler/parcel#9708] [Q&A]
I'm developing on RPi5 (linux_arm) and use elm-tooling to install elm, elm-format, and elm-json.

Installed parcel via npm and coded the Elm example from the doc.

Called parcel via npx which barfed on installing @parcel/transformer-elm.  Worse, node-modules/.bin/elm symbolic link had been trampled on.

It would seem that parcel is not recognising that elm compiler is installed and it trying to download it but cannot find an image for linux_arm platform.

How does parcel check for the presence of the elm compiler ?

-- 
Mark Thornber

**Comment by mischnic** [comment]
It just installs the `elm` package from npm: https://github.com/parcel-bundler/parcel/blob/eb548207fe291745cc62f07262fcf83c0c618b63/packages/transformers/elm/package.json#L33

**Reply by emthornber** [reply]
Agreed - and that works on x64 Ubuntu.  My question is why parcel doesn't recognise that the elm compiler is already installed.

-- 
MarkT

**Comment by emthornber** [comment]
Niklas,

… and there is the problem.  NPM does not have a build for arm processor but there is an elm-tooling offering that _does_ support arm.  I’ve installed that and the package creates symbolic links from the place that npm would have installed elm.  However parcel queries npm for elm, finds it doesn’t know about elm (just elm-tooling) and tries to install an unusable elm.

Ideally I’d like parcel to look for both elm-tooling and elm before attempting an installation.


Stay Safe
— 
Mark Thornber

==================== 
E M Thornber CEng MIET 
Enchanted Systems Limited 
Software Toolsmiths 
+44 1503 272097 

Registered in England No: 3595651 
Registered Office: 
Little Garth, Talland Hill 
Polperro, LOOE, Cornwall 
PL13 2JL 
VAT No: GB 717 7967 83 

On 8 May 2024 at 13:49:15, Niklas Mischkulnig ***@***.***) wrote:

It just installs the elm package from npm: https://github.com/parcel-bundler/parcel/blob/eb548207fe291745cc62f07262fcf83c0c618b63/packages/transformers/elm/package.json#L33

**Reply by yamadapc** [reply]
We shouldn't have this check - https://github.com/parcel-bundler/parcel/blob/eb548207fe291745cc62f07262fcf83c0c618b63/packages/transformers/elm/src/loadConfig.js#L29

It seems the node wrapper itself will find the binary ; so there's no need to install that package or enforce that the binary is vendored in `node_modules`. https://github.com/rtfeldman/node-elm-compiler/blob/master/src/index.ts#L5

Then this also shouldn't be passed in https://github.com/parcel-bundler/parcel/blob/v2/packages/transformers/elm/src/ElmTransformer.js#L109 

Alternatively the upstream elm package should be fixed.
